.github/workflows/build-make-check.yml

```diff
@@ -0,0 +1,28 @@
+name: "Build with warnings and make check"
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  makecheck:
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      id: dependencies
+      uses: ./.github/actions/install-dependencies
+
+    - name: Configure sssd
+      uses: ./.github/actions/configure
+
+    - name: Build with warnings and run make check
+      working-directory: x86_64
+      run: |
+        source ../contrib/fedora/bashrc_sssd
+        chmake
```

Comment:
IMHO it would be better to run `make`, `make check` as separate steps so it is clear where we failed. And please, also include `make distcheck`.

We can also store config.h/config.log as artifact.